### PR TITLE
Add an Execute on Binder button

### PR DESCRIPTION
**Pull request by parente**
Adds launch on binder support to the github and gist providers.
Things to get done before merging:

* [x] Make the executor URL a parameter (disable by default, enable in the public nbviewer)
* [x] Switch to CSS for the icon
* [x] Colorize icon on hover (for consistency with the other icons)
* [x] Add the execute icon to a github / gist tree browser page too
* [x] Fix examples we own on the frontpage (!)
* [x] More regular rotation of front page examples (announce https://goo.gl/forms/WayjU9VW7MYvKSb12 on the mailing list)

#792 captures some things to consider beyond this first PR.

nbviewer-binder-button

### CODE REVIEW

**Comment by parente** [comment]
@minrk Ready for you to have a look if you'd like. Not sure who else to tag from a nbviewer + binder perspective. @betatim? @willingc?

**Comment by minrk** [comment]
This is great! We probably want to allow disabling the Binder links if anyone wants to. Is this possible already? Maybe if binder_base_url is Falsy?

File: nbviewer/providers/base.py [hunk 1]
```diff
@@ -607,7 +611,8 @@ def filter_formats(self, nb, raw):
     def finish_notebook(self, json_notebook, download_url, provider_url=None,
                         provider_icon=None, provider_label=None, msg=None,
                         breadcrumbs=None, public=False, format=None, request=None,
-                        title=None):
+                        title=None, executor_url=None, executor_label=None,
```
Comment (willingc) [inline nbviewer/providers/base.py]: Perhaps add to docstring that `title` is the title of the application that is displayed i.e. nbvviewer, and that it's not the metadata name of a notebook.

**Comment by parente** [comment]
@minrk > This is great! We probably want to allow disabling the Binder links if anyone wants to. Is this possible already? Maybe if binder_base_url is Falsy?

Agreed. I forgot to add that to my todo list after talking with Nick about it during the open studio. I'll add it in when I rebase.

File: nbviewer/providers/base.py (hunk 1 above)
Comment (parente) [reply nbviewer/providers/base.py]: `title` was already declared, but I'm all for tidying up a bit. I can doc all the parameters to that key method while I'm in there.

Comment (willingc) [reply nbviewer/providers/base.py]: Yeah, there was another comment elsewhere in the code that was interchangeably using title and notebook name.

**Comment by parente** [comment]
Addressed review comments and rebased on master.

**Comment by parente** [comment]
> This is the most beautiful and helpful docstring. Thanks.

@willingc ❤️ 🎁  numpydoc format FTW.

**Comment by parente** [comment]
Merging what's here. We can iterate in additional PRs if we find problems.

**Comment by parente** [comment]
I redeployed the public nbviewer.jupyter.org site with the latest docker image containing this change. Seems to be working as expected. If we need to rollback for some reason, the prior docker image ID still present on the hosts is `7de3e2cf6589`.

I plan to work on updating the FAQ and issue templates next to guide people on what to do when a binder doesn't launch properly or a notebook doesn't run in the binder container. The FAQ needs a lot of other updates, so I'll get those while I'm making changes.

**Pull request merged**